ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CheckDetailsTable.tsx

```diff
@@ -38,22 +56,55 @@ function CheckDetailsTable({
     profileName,
     tableState,
     getSortParams,
+    searchFilterConfig,
+    searchFilter,
+    onSearch,
+    onCheckStatusSelect,
 }: CheckDetailsTableProps) {
     const { page, perPage, setPage, setPerPage } = pagination;
 
     return (
         <>
             <Toolbar>
                 <ToolbarContent>
-                    <ToolbarItem variant="pagination" align={{ default: 'alignRight' }}>
-                        <Pagination
-                            itemCount={checkResultsCount}
-                            page={page}
-                            perPage={perPage}
-                            onSetPage={(_, newPage) => setPage(newPage)}
-                            onPerPageSelect={(_, newPerPage) => setPerPage(newPerPage)}
+                    <ToolbarGroup className="pf-v5-u-w-100">
+                        <ToolbarItem className="pf-v5-u-flex-1">
+                            <CompoundSearchFilter
+                                config={searchFilterConfig}
+                                searchFilter={searchFilter}
+                                onSearch={onSearch}
+                            />
+                        </ToolbarItem>
+                        <ToolbarItem>
+                            <CheckStatusDropdown
+                                searchFilter={searchFilter}
+                                onSelect={onCheckStatusSelect}
+                            />
+                        </ToolbarItem>
+                        <ToolbarItem variant="pagination" align={{ default: 'alignRight' }}>
+                            <Pagination
+                                itemCount={checkResultsCount}
+                                page={page}
+                                perPage={perPage}
+                                onSetPage={(_, newPage) => setPage(newPage)}
+                                onPerPageSelect={(_, newPerPage) => setPerPage(newPerPage)}
+                            />
+                        </ToolbarItem>
+                    </ToolbarGroup>
+                    <ToolbarGroup className="pf-v5-u-w-100">
+                        <SearchFilterChips
+                            filterChipGroupDescriptors={[
+                                {
+                                    displayName: 'Cluster',
+                                    searchFilterName: CLUSTER_QUERY,
+                                },
+                                {
+                                    displayName: 'Compliance Status',
+                                    searchFilterName: CHECK_STATUS_QUERY,
+                                },
+                            ]}
                         />
-                    </ToolbarItem>
+                    </ToolbarGroup>
                 </ToolbarContent>
             </Toolbar>
```

Comment:
Is it possible to reuse this toolbar for all of the compliance pages, possibly with some minor customization via props? I see that there are other PRs open with a very similar implementation.